### DISCUSSION TITLE
mysql-client on macOS Catalina 10.15.7 not working when attempting a connection from R

### DISCUSSION THREAD

**Discussion by alexdiazortiz** [Homebrew/discussions#281] [Casks]
Hi there,

I am interested on installing the ODBC MySQL driver on macOS Catalina 10.15.7 since I wan to connect to a MySQL database  from R via the `DBI` and `odbc` packages.

I have installed `unixodbc` and it's functional as I have tested it with a different set of drivers.

The bottle install of `mysql-client`  seems to be uneventful (see log file attached), 

[brew_bottle_install_mysql-client.log](https://github.com/Homebrew/discussions/files/5684502/brew_bottle_install_mysql-client.log)

yet when attempting the connection (from R) it breaks with some cryptic error message 

`Error: nanodbc/nanodbc.cpp:1021: 00000: [unixODBC][Driver Manager]Driver's SQLAllocHandle on SQL_HANDLE_HENV failed`

I have declared the driver in the `odbcinst.ini` as follows:

`Driver = /usr/local/opt/mysql-client/lib/libmysqlclient.dylib`

as suggested by the `mysql-client` caveat of being keg-only. 

It seems as if the driver is not compiled against the `unixodbc` driver manager.

Am I missing something?

**Comment by gromgit** [comment]
I'm pretty sure you need to use a _MySQL ODBC driver_ (which exports the ODBC API) rather than the MySQL client driver (which exports the MySQL API).

A quick search reveals that MySQL has their own ODBC driver ([Connector/ODBC](https://dev.mysql.com/downloads/connector/odbc/)), but if that doesn't work for you, an online search for "mysql odbc driver macos" may yield the results you desire.

**Reply by alexdiazortiz** [reply]
@gromgit  Thanks for your feedback.
